tests/apps/test_similarity.py

```diff
@@ -293,6 +291,24 @@ def test_odd_photon_numbers(self, k):
         graph = nx.complete_graph(10)
         assert similarity.prob_orbit_exact(graph, [k]) == 0.0
 
+    def test_correct_result_returned(self, monkeypatch):
+        """Tests if the call to _get_state function is performed correctly and probabilities over all
+        permutations of the given orbit are summed correctly. The test monkeypatches the fock_prob
+        function so that the probability is the same for each sample permutation and
+        is equal to 1/8. For a 4-mode graph, [1, 1] has 6 possible permutations. """
+        graph = nx.complete_graph(4)
+        with monkeypatch.context() as m:
+            m.setattr("strawberryfields.backends.gaussianbackend.states.GaussianState.fock_prob",
+                      lambda *args, **kwargs: 1/8)
+            assert similarity.prob_orbit_exact(graph, [1, 1]) == 6/8
+
+    def test_known_result(self):
```

Comment:
This test seems dangerous: how small exactly is the orbit probability? Also, it seems you can pass this test with a wrong implementation of the function, e.g., if it's not summing correctly over all samples in the orbit -- if the probabilities are small it doesn't matter if you miss a few in the sum.

I think a better version of this test would be to set a lower number of photons, say the orbit `[2]`, a higher mean photon number, say `n_mean = 2`, then calculate the probability by hand, hard-code it into the test, and check that we get the right answer